src/main/java/com/redhat/devtools/lsp4ij/usages/LSPFindActionTracker.java

```diff
@@ -0,0 +1,76 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package com.redhat.devtools.lsp4ij.usages;
+
+import com.intellij.find.actions.FindUsagesAction;
+import com.intellij.openapi.actionSystem.AnAction;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.actionSystem.CommonDataKeys;
+import com.intellij.openapi.actionSystem.ex.AnActionListener;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiFile;
+import com.intellij.usages.UsageTargetUtil;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * "Find Usages" action execution tracker used to know if {@link LSPUsageTargetProvider}
+ * must return LSP {@link com.intellij.usages.UsageTarget} or not:
+ *
+ * <ul>
+ *     <li>when language doesn't return some UsageTarget, LSP UsageTarget must be returned to start the find usages.</li>
+ *     <li>when language (ex: Java) return some UsageTarget, LSP UsageTarget must be not collected to avoid

```

Comment:
when a UsageTarget is already associated to a language (e.g. Java in IDEA), the LSP UsageTarget must not be returned, to avoid overriding the platform native support for `Find usages` for that language.

Reply:
fixed